client/my-sites/migrate/section-migrate.jsx

```diff
@@ -76,11 +87,20 @@ class SectionMigrate extends Component {
 			 * Note this migrationStatus is local, thus the setState vs setMigrationState.
 			 * Call to updateFromAPI will update both local and non-local state.
 			 */
-			this.setState( { migrationStatus: 'inactive', errorMessage: '' }, this.updateFromAPI );
+			this.setState(
+				{
+					migrationStatus: 'inactive',
+					errorMessage: '',
+				},
+				this.updateFromAPI
+			);
 		} );
 	};
 
 	setMigrationState = state => {
+		if ( 'error' === this.state.migrationStatus ) {
+			return;
+		}
```

Comment:
This is necessary to avoid a response to a request to the `migration-status` endpoint accidentally resetting the state after a response from the `migrate/from` endpoint has set it to `error`.

Reply:
Could you add a comment explaining that? I think that would be helpful, as I'd be confused if I were to run into this in the future.

Reply:
Also, how does an error state from the server that occurred after the initial API call to start the migration get cleared? Possible that I'm just missing something.

Reply:
Thanks, comment added. 

When either the `migrate/from` or `migration-status` endpoints return an error, we show the error view with the "Back to your site" button. Clicking this calls `resetMigration`, which updates the local state with `this.setState`, bypassing `setMigrationState`. So users should still be able to reset an error.